<a href="https://colab.research.google.com/github/Ksenia-90/probability_theory/blob/lesson_5/lesson_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование**

In [3]:
import numpy as np
from scipy import stats
import numpy as np

**Задача 1**

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания mu с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

In [28]:
t_obj = stats.t(256 - 1)
t = t_obj.ppf(1 - 0.05 / 2)
t

1.9693105698491928

In [36]:
s = 16
n = 256
m = 80
print ('Доверительный интервал:', (m - t * s / np.sqrt(n), m + t * s / np.sqrt(n)))

Доверительный интервал: (78.0306894301508, 81.9693105698492)


**Задача 2**

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190


Известно, что их веса распределены нормально.
1. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?
2. Найдите P-значение для данного теста.

Проверим утвержденеи продавца, что средний вес пачки печенья составляет 200 г.

In [41]:
weight  = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
m = weight.mean()
print(m)

198.5


In [42]:
n = 10
sigma = np.sqrt(((weight - m)**2).sum()/(n-1))
print ('sigma', sigma)

sigma 4.453463071962462


Табличное значение коэффициента Стьюдента

In [43]:
t_tabl = 3.250

In [44]:
t = stats.t(n-1)
t.cdf(3.250)

0.9950013154579892

In [45]:
print ('Доверительный интервал равен:', (m - t_tabl*sigma/np.sqrt(n), m + t_tabl*sigma/np.sqrt(n)))

Доверительный интервал равен: (193.9229967955732, 203.0770032044268)


Доверительный интервал, построенный с учётом вероятности 99,5%, включает значение предполагаемого среднего значения 200г., следовательно утверждение продавца, что средний вес пачки = 200 гр, верно.

**Задача 3**

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

выдвинем 2 гипотизы:
*  $H_0$ : средний диаметр шарика для подшипникак 17 мм
*  $H_1$ : средний диаметр шарика для подшипникак больше 17 мм



Предполагается, что количество шариков большое, то распределение диаметра шариков является нормальным или близким к нормальному. Найдем sigma

In [5]:
n = 100
D = 4
sigma = np.sqrt(D/n)
sigma

0.2

установим уровень значимости 𝛼

In [21]:
alpha = 0.05

Выбранный уровень значимости $\alpha = 0.025$ означает, что для проверки гипотезы о среднем значении alpha = 17 требуется, чтобы выборочное среднее попадало в правосторонний интервал, который равен $a - 2 \cdot \sigma,  a + 2 \cdot \sigma$ (по правилу 2-х сигм).



Kритическая область лежит правее точки $a + 2 \cdot \sigma $ и если выборочное среднее 17.5 mm попадёт в критическую область, то  гипотеза $H_0$ отвергается.

In [46]:
a = 17
sigma = 0.2
(a + 2 * sigma)

17.4

Доверительный интервал $\left( 17.4, \infty \right)$ Гипотеза о том, что шарики для подшипников имеют средний диаметр 17 мм, при выбранном уровне значимости $\alpha = 0.025$ отвергается.



In [22]:
stats.norm.ppf(1 - 0.025, 17, 0.2)

17.39199279690801

**Задача 4 (Дополнительно)**

Дана выборка диаметров подшипников из примера 1 с занятия 5:

samples = [0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02 ]

Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна 0.0625 при уровне значимости alpha = 0.05. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: H = (n - 1) * sample_variance / variance, где n - число элементов в выборке, sample_variance - несмещённая оценка дисперсии, variance - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром df = n - 1. Её квантили можно найти с помощью функции scipy.stats.chi2.ppf.

In [4]:
samples = np.array([0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
                    1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
                    0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ])
alpha = 0.05
D = 0.0625
n = 24 

In [5]:
sigma = np.sqrt(D/n)
sigma

0.05103103630798288

In [8]:
mean_ = samples.mean()
print(mean_)

1.1084541666666665
